core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -768,6 +770,21 @@ public PlanOptimizers(
                             // Must run before AddExchanges and after ReplicateSemiJoinInDelete
                             // to avoid temporarily having an invalid plan
                             new DetermineSemiJoinDistributionType(costComparator, taskCountEstimator)))));
+
+            builder.add(new IterativeOptimizer(
+                    ruleStats,
+                    statsCalculator,
+                    estimatedExchangesCostCalculator,
+                    ImmutableSet.<Rule<?>>builder()
+                            .addAll(pushIntoTableScanRulesExceptJoins)
+                            // Must run before DetermineTableScanNodePartitioning because PushJoinIntoTableScan creates new table handles
```

Comment:
this comment is no longer valid (there is no order in `IterativeOptimizer`). Comment below is sufficient

Reply:
Done.